segway/input_master.py

```diff
@@ -518,334 +504,109 @@ def make_dirichlet_table(self):
-            # to fix the number of mixture components to one
-            yield "dpmf_always 1 1.0"
-        # Otherwise
-        else:
-            # Create a dense probability mass function of dirichlet constants
-            # with the same amount of mixture components
-            object_tmpl = "dpmf_${seg}_${subseg}_${track} ${num_mix_components} "\
-                        "DirichletConst %s ${weights}" % GAUSSIAN_MIXTURE_WEIGHTS_PSEUDOCOUNT
-            component_weight = str(round(1.0 / self.num_mix_components,
-                                         ROUND_NDIGITS))
-            weights = (" " + component_weight) * self.num_mix_components
-            substitute = Template(object_tmpl).substitute
-            data = self.make_data()
-            for mapping in self.generate_tmpl_mappings():
-                mapping["weights"] = weights
-                track_index = mapping["track_index"]
-                mapping["num_mix_components"] = self.num_mix_components
-                if self.distribution == DISTRIBUTION_GAMMA:
-                    mapping["min_track"] = self.get_track_lt_min(track_index)
-
-                if data is not None:
-                    seg_index = mapping["seg_index"]
-                    subseg_index = mapping["subseg_index"]
-                    mapping["datum"] = data[seg_index, subseg_index, track_index]
-                yield substitute(mapping)
 
 class VirtualEvidenceSpec(ParamSpec):
     type_name = "VE_CPT"
 
     # According to GMTK specification (tksrc/GMTK_VECPT.cc)
-    # this should be of the format: 
+    # this should be of the format:
     # CPT_name num_par par_card self_card VE_CPT_FILE
     # nfs:nfloats nis:nints ... fmt:obsformat ... END
     object_tmpl = "seg_virtualEvidence 1 %s 2 %s nfs:%s nis:0 fmt:ascii END"
     copy_attrs = ParamSpec.copy_attrs + ["virtual_evidence", "num_segs"]
 
     def make_virtual_evidence_spec(self):
-        return self.object_tmpl % (self.num_segs, VIRTUAL_EVIDENCE_LIST_FILENAME, self.num_segs)
+        return self.object_tmpl % (
+        self.num_segs, VIRTUAL_EVIDENCE_LIST_FILENAME, self.num_segs)
 
     def generate_objects(self):
         yield self.make_virtual_evidence_spec()
-
+  
 
 class InputMasterSaver(Saver):
     resource_name = "input.master.tmpl"
     copy_attrs = ["num_bases", "num_segs", "num_subsegs",
                   "num_track_groups", "card_seg_countdown",
                   "seg_countdowns_initial", "seg_table", "distribution",
-                  "len_seg_strength", "resolution", "random_state", "supervision_type",
+                  "len_seg_strength", "resolution", "random_state",
+                  "supervision_type",
                   "use_dinucleotide", "mins", "means", "vars",
                   "gmtk_include_filename_relative", "track_groups",
-                  "num_mix_components", "virtual_evidence"] 
+                  "num_mix_components", "virtual_evidence", "tracks"]
```

Comment:
why was `tracks` added when it was necessary before to create the input master?

Reply:
`tracks` is being used in the initializer of ParamSpec. Previously, there was no instance of ParamSpec in InputMasterSaver, so this attribute was not needed.